packages/enzyme/src/ShallowWrapper.js

```diff
@@ -635,6 +635,20 @@ class ShallowWrapper {
               instance.state,
             );
           }
+          if (
+            shouldRender
+            && instance
+            && context
+          ) {
+            if (lifecycles.componentWillReceivePropsOnShallowRerender) {
```

Comment:
can you elaborate on this lifecycle option?

Reply:
> This code corrects shallow()'s calling of componentWillReceiveProps() and UNSAFE_componentWillReceiveProps() under react 17.

-- https://github.com/enzymejs/enzyme/pull/2534/files#r687066582